osu.Game/Screens/Play/ScreenWithBeatmapBackground.cs

```diff
@@ -0,0 +1,61 @@
+﻿// Copyright (c) 2007-2018 ppy Pty Ltd <contact@ppy.sh>.
+// Licensed under the MIT Licence - https://raw.githubusercontent.com/ppy/osu/master/LICENCE
+
+using osu.Framework.Allocation;
+using osu.Framework.Configuration;
+using osu.Framework.Graphics;
+using osu.Game.Configuration;
+using osu.Game.Screens.Backgrounds;
+using OpenTK;
+
+namespace osu.Game.Screens.Play
+{
+    public abstract class ScreenWithBeatmapBackground : OsuScreen
+    {
+        protected override BackgroundScreen CreateBackground() => new BackgroundScreenBeatmap(Beatmap);
+
+        public override bool AllowBeatmapRulesetChange => false;
+
+        protected const float BACKGROUND_FADE_DURATION = 800;
+
+        protected float BackgroundOpacity => 1 - (float)DimLevel;
+
+        #region User Settings
+
+        protected Bindable<double> DimLevel;
+        protected Bindable<double> BlurLevel;
+        protected Bindable<bool> ShowStoryboard;
+        protected Bindable<bool> MouseWheelDisabled;
+        protected Bindable<double> UserAudioOffset;
+
+        #endregion
+
+        [BackgroundDependencyLoader]
+        private void load(OsuConfigManager config)
+        {
+            DimLevel = config.GetBindable<double>(OsuSetting.DimLevel);
+            BlurLevel = config.GetBindable<double>(OsuSetting.BlurLevel);
+            ShowStoryboard = config.GetBindable<bool>(OsuSetting.ShowStoryboard);
+
+            MouseWheelDisabled = config.GetBindable<bool>(OsuSetting.MouseDisableWheel);
+
+            UserAudioOffset = config.GetBindable<double>(OsuSetting.AudioOffset);
+        }
+
+        protected void ConfigureBackgroundUpdate()

```

Comment:
why is this a thing? it's called from the same place in both consumers of this screen. is there something i'm missing or can this just be moved to `OnEntering`?

Reply:
You are correct. Can be moved to `ScreenWithBeatmapBackground.OnEntering`.